Demo/src/main/java/com/paypal/android/ui/card/DateString.kt

```diff
@@ -58,6 +60,27 @@ data class DateString(private val rawDate: String) {
         this.didPadZero = padZero
         this.didAddSlash = addSlash
         this.formatted = formatted
+
+        // TODO: handle invalid date string
+        var formattedMonth = ""
+        var formattedYear = ""
+
+        val dateStringComponents = formatted.split("/")
+        if (dateStringComponents.isNotEmpty()) {
+            formattedMonth = dateStringComponents[0]
+            if (dateStringComponents.size > 1) {
+                val rawYear = dateStringComponents[1]
+                formattedYear = if (rawYear.length == 2) {
+                    // pad with 20 to assume 2000's
+                    "20$rawYear"
+                } else {
+                    rawYear
+                }
```

Comment:
Also take it or leave it: seems like we could single line this which is close to the ternary operator in Swift. Bummer though it's not quite as clean as the Swift counterpart ([Kotlin docs here](https://kotlinlang.org/docs/control-flow.html#if-expression)).

Reply:
Yeah that's a good call. I went ahead and flattened it out here: [1b5a4ae](https://github.com/paypal/paypal-android/pull/204/commits/1b5a4aeee93f860b832568904e79a9c7aff9536f)